ionoscloud/resource_server.go

```diff
@@ -1026,6 +1031,27 @@ func resourceServerUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 	return resourceServerRead(ctx, d, meta)
 }
 
+func deleteAllVolumes(ctx context.Context, d *schema.ResourceData, client *ionoscloud.APIClient, server *ionoscloud.Server) diag.Diagnostics {
```

Comment:
@gabi-ionos not sure how you tested this, but it's not working. I created a server using the configuration presented in the [issue](https://github.com/ionos-cloud/terraform-provider-ionoscloud/issues/379) and then I manually added two volumes using the `DCD`. After this, I removed the server from the `tf` plan and, after deletion, there were dangling volumes in the `DCD`.